### DISCUSSION TITLE
Alternative to the Shelly power meetering

### DISCUSSION THREAD

**Discussion by EDSTOBI** [arendst/Tasmota#16386] [Q&A] [answered]
Hello
I use several Shelly to measure the current of my heating system.
I have been having problems for days which I have not been able to explain.
Changing the Shelly among themselves has not brought any improvement. Switching to a Shelly plus showed me that the Shelly goes into temperature.

The ambient temperature in the heating room is 31 degrees.
30-40 watts run through the Shelly. It cannot cope with that

Do you know an alternative to the Shelly? Switching is not important, I need the consumption of the system 4 pumps and a total system incl. burner.

EDSTOBI

**Comment by sfromis** [comment, marked as answer]
Lots of power monitoring devices in https://templates.blakadder.com/2nice-UP111.html and https://templates.blakadder.com/relay.html

You did not post what Shelly device you see as troublesome (or how you see that), but I've seen no signs of trouble with Shelly 2.5 and Shelly Plus 1PM/2PM. Had odd results with the Shelly Dimmer 2 for energy monitoring.

**Comment by arendst** [comment]
> Shelly goes into temperature

What exactly happens? If Tasmota powers off the device based on a "suspected temperature of over 90C" this is likely caused by the default settings and the not very reliable ESP32 temperature sensor. In that case disable the ESP32 temp sensor with command ``setsensor127 0`` or raise the overtemp value of default 90 to something about 120 using command ``SetOption42 120``.

**Reply by EDSTOBI** [reply]
thank you for the answer. 
I have just extended the monitoring and watch how the room temperatures mooves to 30 degree and Shelly 1 plus PM is slowly pushing towards the 90 degrees.
I already have the laser thermometer ready to measure the real temperature with housing (later without housing). (I don't feel like burning down my house). 

I will describe the outcomming here

I will replace the current measurement via shelly with another technique in the short term. I will start my internet research as soon as I have time.

I don't have a good feeling about using the shelly in warm environments. All the solar and heating technology is under the roof, and once the heat is there, I can't get it out. 1000 litres of 90 degree water will always radiate heat, you can insulate it all you want.

**Reply by sfromis** [reply]
You are still not very specific about details. If the Temperature you see "pushing towards the 90 degrees" is that "ESP32 Temperature", I'd suggest you to follow the suggestion from arendst about disabling that "not very reliable" sensor using `SetSensor127 0`. It is just not a working internal temperature sensor.

On a Shelly Plus 1PM (without a load), I'm seeing `"ESP32":{"Temperature":87.78}` while the device is not even lukewarm to the touch. It would be nice if that temperature sensing was working, but it is not. Get rid of the value. Unless you've determined for real that the module is getting hot, I'd just not at all worry about that fake number.

Technically, the hardware includes another temperature sensor available on analog, but without having working values for `AdcParam` to match the hardware, that has not been included in the template.

**Reply by EDSTOBI** [reply]
Hello
unfortunately it is no longer warm enough to reproduce this error.
But today the Shelly plus 1PM reported 85.7 degrees, the enclosure was 41 degrees on one side and 37 degrees on the other side.
So I can safely change the value to 120 degrees.
Thanks for this tip.

**Reply by sfromis** [reply]
Is it that `ESP32 Temperature` value you see. **Stop using that**, and disable the misleading number with `SetSensor127 0`. The number is IME so useless that there is no point in changing the upper limit to try to accommodate it.

**Comment by EDSTOBI** [comment]
SetSensor127 0 is now implemented in all my shelly
thanks for your support
